lib/omnibus/publisher.rb

```diff
@@ -61,7 +66,50 @@ def initialize(pattern, options = {})
     # @return [Array<String>]
     #
     def packages
-      @packages ||= FileSyncer.glob(@pattern).map { |path| Package.new(path) }
+      @packages ||= begin
+        publish_packages = Array.new
+        build_packages   = FileSyncer.glob(@pattern).map { |path| Package.new(path) }
+
+        if @options[:platform_mappings]
+          # the platform map is a simple hash with publish to build platform mappings
+          @options[:platform_mappings].each_pair do |build_platform, publish_platforms|
+            # Splits `ubuntu-12.04` into `ubuntu` and `12.04`
+            build_platform, build_platform_version = build_platform.split('-')
```

Comment:
Awesome tip @danielsdeleo, I think `String#rpartition` is just what is needed. If there is an extra hyphen it will most likely appear in the platform.